zkvm/docs/zkvm-spec.md

```diff
@@ -828,6 +787,54 @@ V == v·B + 0·B2
+
+1. The alternative clauses `{C_i}`, each of which is a [program](#program), and their corresponding blinding factors are stored as [blinded programs](#blinded-program) and compressed in a [Merkle tree](#merkle-binary-tree) with root `M`.
+2. A signing key `X` and the Merkle root `M` (from 1) are committed to a single signing key `P` using a hash function `h1`, such that `P = X + h1(X, M)`. This makes signing for `P` possible if parties want to sign for `X` and avoids revealing the alternative clauses.
+3. Calling a program will check a [call proof](#call-proof) to verify the program's inclusion in the Taproot tree before executing the program.
+
+The commitment scheme is defined using the [transcript](#transcript) protocol by committing (1) `X`, the signing key, as a compressed 32-byte point, and (2) `M`, the root of the Merkle tree constructed from the `n` programs and (3) squeezing a scalar bound to all `n` programs.
+
+```
+// given Merkle root M and signing key X
+t = Transcript("ZkVM.taproot")
+t.commit_bytes(b"key", X); // Compressed Ristretto
+t.commit_bytes(b"merkle", M); // [u8; 32]
+t.challenge_scalar(b"h")
+```
+
+Typically, to select a program to run from a leaf of the Merkle tree, we must navigate branches from the Merkle root. We tweak this procedure: the Merkle root is constructed from a provided Merkle path. It is then used to verify a relation between the contract predicate and the provided program. A branch is chosen using the [`call`](#call) instruction, which takes a program, [call proof](#call-proof), and contract. The [predicate](#predicate) `P` is read from the contract. The call proof contains the signing key `X`, a list of left-right neighbors, and the Merkle path. The neighbor list, Merkle path, and the provided program are used to reconstruct the Merkle root `M`. If the relation `P = X + h1(X, M)` is verified, then the associated program will be executed.
+
+
+### Call proof
+
+This struct is an argument to the [`call`](#call) instruction. It contains the information to verify that a program is represented in the [Taproot](#taproot) Merkle tree, in the following fields:
+ - `X`, the signing key
+ - `neighbors`, an array of left-right neighbors in the Merkle tree. Each is a LE32 item representing the hash of its children. It is the neighbor of an item on the Merkle path to the leaf hash of the program string. The leaf hash and root hash are not included. Neighbor hashes are ordered from bottom-most to top-most. As an example, consider a Merkle tree with more than three levels. The first element is the neighbor of the leaf hash; the second element is the neighbor of the leaf hash's parent; and the last is a child of the Merkle root.
```

Comment:
> Each is a LE32 item representing the hash of its children.

Hashes are 256-bit-blobs. They are not little- or big-endian. And surely they are not LE32 which is "32-bit little endian integer".

Reply:
D'oh, thanks for catching this! I think this must have been a holdover/copy-paste from something else.